summingbird-storm/src/main/scala/com/twitter/summingbird/storm/FlatMapBoltProvider.scala

```diff
@@ -162,4 +156,15 @@ case class FlatMapBoltProvider(storm: Storm, jobID: JobId, stormDag: Dag[Storm],
       case None => getIntermediateFMBolt[Any, Any].asInstanceOf[BaseBolt[Any, Any]]
     }
   }
+
+  private def inputEdges[Input]: Map[String, Edge[Input]] = {
```

Comment:
Yes, I want to make this logic in a way you said - use same `Edge` instances for input and output with double check on edges compatibility at topology construction time (that's why I left `shardsCount` for example in grouping declaration).